Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py

```diff
@@ -0,0 +1,448 @@
+import demistomock as demisto
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+
+import requests
+import traceback
+from typing import Dict, Any
+import base64
+import json
+from dateparser import parse
+from pytz import utc
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+''' CONSTANTS '''
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+INCIDENT_COLUMNS = ['start_time', 'end_time', 'id', 'incident_type', 'severity',
+                    'ended', 'affected_assets']
+ASSET_COLUMNS = ['status', 'name', 'asset_id', 'last_seen', 'ip_addresses',
+                 'tenant_name']
+INCIDENT_TYPE = 'GuardiCore Incident'
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """
+       Client for GuardiCoreV2
+
+       Args:
+          username (str): The GuardiCore username for API access.
+          password (str): The GuardiCore password for API access.
+          access_token (str): The GuardiCore access token, generated automatically from username and password.
+          base_url (str): The GuardiCore API server URL.
+    """
+
+    def __init__(self, proxy: bool, verify: bool, base_url: str, username: str,
+                 password: str):
+        super().__init__(proxy=proxy, verify=verify, base_url=base_url)
+        self.username = username
+        self.password = password
+        self.access_token = ""
+        self.base_url = base_url
+
+        self.login()
+
+    def login(self):
+        integration_context = get_integration_context()
+
+        if self.is_access_token_valid(integration_context):
+            access_token = integration_context.get('access_token')
+            self.save_access_token(access_token)
+        else:
+            demisto.debug(
+                "GuardicoreV2 - Generating a new token (old one isn't valid anymore).")
+            self.generate_new_token()
+
+    def save_access_token(self, access_token: str):
+        self.access_token = access_token
+        authorization_value = f'bearer {access_token}'
+        self._headers = {
+            "Authorization": authorization_value}
+
+    def is_access_token_valid(self, integration_context: dict):
+        access_token_expiration = integration_context.get('expires_in')
+        access_token = integration_context.get('access_token')
+        demisto.debug(
+            'GuardicoreV2 - Checking if context has valid access token...'
```

Comment:
I suggest having an `INTEGRATION_NAME = 'Guardicore v2'` for all of the debug messages.